mirdata/download_utils.py

```diff
@@ -246,10 +247,28 @@ def extractall_unicode(zfile, out_dir):
     for m in zfile.infolist():
         data = zfile.read(m)  # extract zipped data into memory
 
-        if m.filename.encode("cp437").decode() != m.filename.encode("utf8").decode():
-            disk_file_name = os.path.join(out_dir, m.filename.encode("cp437").decode())
+        ### get filename
+        name = m.filename
+        try:
+            ### non-utf encoding
+            filename = name.encode("cp437")
+        except UnicodeEncodeError:
+            ### utf encoding
+            filename = name.encode("utf8")
+
+        ### check for irmas filename encoding
+        if filename.decode(errors="ignore") != name.encode("utf8").decode():
+            filename = name.encode("cp437")
+
+        ### detect encoding
+        encoding = chardet.detect(filename)["encoding"]
```

Comment:
I'm a bit confused by the logic - is this needed? It looks like we should know the encoding based on the result of the try/except